src/main/java/org/gridsuite/study/server/service/SecurityAnalysisService.java

```diff
@@ -225,61 +215,114 @@ public void invalidateSaStatus(List<UUID> uuids) {
+    public UUID duplicateSecurityAnalysisParameters(UUID sourceParametersUuid) {
+        Objects.requireNonNull(sourceParametersUuid);
+
+        var path = UriComponentsBuilder.fromPath(DELIMITER + SECURITY_ANALYSIS_API_VERSION + "/parameters/{sourceParametersUuid}")
+                .buildAndExpand(sourceParametersUuid).toUriString();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<Void> httpEntity = new HttpEntity<>(null, headers);
+
+        try {
+            return restTemplate.exchange(securityAnalysisServerBaseUri + path, HttpMethod.POST, httpEntity, UUID.class).getBody();
+        } catch (HttpStatusCodeException e) {
+            throw handleHttpError(e, CREATE_SECURITY_ANALYSIS_PARAMETERS_FAILED);
+        }
+    }
+
+    public String getSecurityAnalysisParameters(UUID parametersUuid) {
+        Objects.requireNonNull(parametersUuid);
+        String parameters;
+
+        String path = UriComponentsBuilder.fromPath(DELIMITER + SECURITY_ANALYSIS_API_VERSION + "/parameters/{uuid}")
+                .buildAndExpand(parametersUuid).toUriString();
+
+        try {
+            parameters = restTemplate.getForObject(securityAnalysisServerBaseUri + path, String.class);
+        } catch (HttpStatusCodeException e) {
+            if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
+                throw new StudyException(SECURITY_ANALYSIS_PARAMETERS_NOT_FOUND);
+            }
+            throw handleHttpError(e, GET_SECURITY_ANALYSIS_PARAMETERS_FAILED);
+        }
+        return parameters;
+    }
+
+    public UUID getSecurityAnalysisParametersUuidOrElseCreateDefaults(StudyEntity studyEntity) {
+        if (studyEntity.getSecurityAnalysisParametersUuid() == null) {
+            studyEntity.setSecurityAnalysisParametersUuid(createDefaultSecurityAnalysisParameters());
+
+        }
+        return studyEntity.getSecurityAnalysisParametersUuid();
     }
 
-    public static SecurityAnalysisParametersValues getDefaultSecurityAnalysisParametersValues() {
-        return SecurityAnalysisParametersValues.builder()
-                .lowVoltageAbsoluteThreshold(DEFAULT_LOW_VOLTAGE_ABSOLUTE_THRESHOLD)
-                .lowVoltageProportionalThreshold(DEFAULT_LOW_VOLTAGE_PROPORTIONAL_THRESHOLD)
-                .highVoltageAbsoluteThreshold(DEFAULT_HIGH_VOLTAGE_ABSOLUTE_THRESHOLD)
-                .highVoltageProportionalThreshold(DEFAULT_HIGH_VOLTAGE_PROPORTIONAL_THRESHOLD)
-                .flowProportionalThreshold(DEFAULT_FLOW_PROPORTIONAL_THRESHOLD)
-                .build();
+    public void deleteSecurityAnalysisParameters(UUID uuid) {
+        Objects.requireNonNull(uuid);
+        String path = UriComponentsBuilder.fromPath(DELIMITER + SECURITY_ANALYSIS_API_VERSION + "/parameters/{parametersUuid}")
+                .buildAndExpand(uuid)
+                .toUriString();
+
+        restTemplate.delete(securityAnalysisServerBaseUri + path);
```

Comment:
I know it's not done in voltage init but I'm wondering why @ayolab @SlimaneAmar

Reply:
What would you do with that exception ? Since it's only called on study deletion we don't really care if it fails. The study will be deleted anyway, we don't block the study deletion if related data deletion fail, it's fire and forget mode (and yes it can create orphans but we already handle the study deletion that way for everything else).